### PR TITLE
Add CI checks that we can serde all benchmark queries

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

N/A

# Rationale for this change

Ballista relies on the ability to serde all benchmark queries. These CI checks help ensure that DataFusion continues to support this.

# What changes are included in this PR?

- Add serde tests for all benchmark queries
- ignore test for q16 for now due to https://github.com/apache/arrow-datafusion/issues/3820 (this will be fixed in https://github.com/apache/arrow-datafusion/pull/4050)

# Are there any user-facing changes?

No

### CODE REVIEW

**Comment by andygrove** [comment]
@avantgardnerio fyi

File: benchmarks/src/bin/tpch.rs [hunk 1]
```diff
@@ -618,291 +630,412 @@ mod tests {
...
+            execute_query(&ctx, query, false).await?;
+        }
+
+        Ok(())
+    }
+}
+
+/// CI checks
+#[cfg(test)]
+#[cfg(feature = "ci")]
```
Comment (alamb) [inline benchmarks/src/bin/tpch.rs:769]: Why feature gate this with `ci`? I would have thought

```rust
#[cfg(test)]
```

was sufficient. 
 
Additionally, I couldn't find evidence that these new tests were actually run:


```shell
-*- mode: grep; default-directory: "~/Downloads/logs_52324/" -*-
Grep started at Tue Nov  1 12:19:52

rg -n -H --no-heading -e 'serde_q7' 

Grep finished with no matches found at Tue Nov  1 12:19:52
```

🤔

**Review by alamb — DISMISSED** [review]
I couldn't find evidence that this test actually ran 🤔

File: benchmarks/src/bin/tpch.rs (hunk 1 above)
Comment (andygrove) [reply benchmarks/src/bin/tpch.rs:769]: These tests rely on the data files existing in a path from `TPCH_DATA` env var, which is set in CI

**Comment by andygrove** [comment]
> I couldn't find evidence that this test actually ran thinking

Thanks for catching that! I pushed a commit to add these tests to the workflow

File: benchmarks/src/bin/tpch.rs (hunk 1 above)
Comment (andygrove) [reply benchmarks/src/bin/tpch.rs:769]: I did look into using MemTable but we don't support serde with MemTable

**Comment by andygrove** [comment]
Verified that tests ran this time:

```
 test ci::serde_q1 ... ok
test ci::serde_q10 ... ok
test ci::serde_q11 ... ok
test ci::serde_q12 ... ok
test ci::serde_q13 ... ok
test ci::serde_q14 ... ok
test ci::serde_q15 ... ok
test ci::serde_q16 ... ignored
test ci::serde_q17 ... ok
test ci::serde_q18 ... ok
test ci::serde_q19 ... ok
test ci::serde_q2 ... ok
test ci::serde_q20 ... ok
test ci::serde_q21 ... ok
test ci::serde_q22 ... ok
test ci::serde_q3 ... ok
test ci::serde_q4 ... ok
test ci::serde_q5 ... ok
test ci::serde_q6 ... ok
test ci::serde_q7 ... ok
test ci::serde_q8 ... ok
test ci::serde_q9 ... ok
```

**Pull request merged**